sdk/core/azure-core/CMakeLists.txt

```diff
@@ -125,7 +123,7 @@ else()
 endif()
 
 if(BUILD_TRANSPORT_CURL)
-  target_link_libraries(azure-core PRIVATE CURL::libcurl)
+  target_link_libraries(azure-core PUBLIC CURL::libcurl)
```

Comment:
Doing this exposes all of CURL to the user when really all we are after is curl_global_init.  We are better served to expose an Azure::Initialize(options) which internally calls global init then to expose the entire curl stack.

Reply:
I am not that familiar with the subject, @vhvb1989, can you comment?

Reply:
Also, would it be possible to eliminate the need for customer to call curl_global_init() altogether?

Reply:
@RickWinter, let's discuss this later today. On Friday we were discussing this, and it was assigned a high priority for the upcoming release, and these concerns were not raised during that discussion.

Reply:
>when really all we are after is curl_global_init.

We can't do this. Imagine we are consumed by a 3rd party library A, and A is consumed by another one.  Do they all need to bubble up the Azure::Initialize(option) ??

Last time we talk about this, we agreed that, the the curl_global_init() must be called by the application owner directly. 

The changes on this PR, however, have nothing related to the curl `global_init`.  What the changes do is to declare that we are including the libcurl headers in source in also in headers (PUBLIC) and not only in source (PRIVATE).
This means that anyone selecting `BUILD_TRANSPORT_CURL` on cmake generate would depend on libcurl, they need to make sure libcurl is available (vcpkg or installed).
Then, they don't need to explicitly link libcurl to their application after linking the Azure SDK because the SDK is linking to it already, so if they use our headers that includes libcurl, they get it for free.

I was against this on the past because I would expect end users to explicitly link libcul all the times if they want to use the Azure SDK, but this is causing VCKPG issues, so I just let it be :)

Reply:
Is there a versioning concern with this? What happens if the customer already has installed and is using an older version of libcurl, which isn't compatible with our SDK?

Reply:
> What happens if the customer already has installed and is using...

CMake will fail during generation step, saying libcurl min version = X.X was not found